web-bundle/src/main/java/com/graphhopper/resources/SPTResource.java

```diff
@@ -62,11 +60,15 @@ public Response doGet(
             throw new IllegalArgumentException("point parameter cannot be null");
 
         StopWatch sw = new StopWatch().start();
-
-        if (!encodingManager.hasEncoder(vehicle))
-            throw new IllegalArgumentException("vehicle not supported:" + vehicle);
-
-        FlagEncoder encoder = encodingManager.getEncoder(vehicle);
+        HintsMap hintsMap = new HintsMap();
+        RouteResource.initHints(hintsMap, uriInfo.getQueryParameters());
+        hintsMap.put(Parameters.CH.DISABLE, true);
```

Comment:
Is it fine to overwrite ch/lm.disable here or should we throw an error?

Reply:
Isn't the none-CH case the default now? But yeah, I think I had to do this for the /custom endpoint too.

Reply:
Hm, no `graphHopper.resolveProfile(hintsMap)` will give us a CH profile or error if it does not exist, so we have to explicitly disable CH/LM to obtain the profile no matter whether a CH/LM profile for these hints exists or not.

Reply:
So, if there is a profile without CH, then this will work without the parameter, but if the profile is listed in `profiles_ch` then we need the disabling parameter?

The problem is, what if we add a new algorithm xy? Then in theory I would expect that the profile selection remains unchanged and we only specify `profiles_xy: [car_xy]`, i.e. the profile `car` without the algorithm xy can still exist. Not sure if this is good as it increases the profile count further.

Reply:
> So, if there is a profile without CH, then this will work without the parameter, but if the profile is listed in profiles_ch then we need the disabling parameter?

Yes, or to put it differently if we put ch.disable=true as we do here we get the profile matching our hints no matter what LM/CH preparations there are (and for isochrones we are not even using them). But you are right the profile selection depends on the available LM/CH preparations (it always did).

> The problem is, what if we add a new algorithm xy? Then in theory I would expect that the profile selection remains unchanged 

But why? If there is a new algorithm we likely want to use it? If the selection remained unchanged what would the new algorithm be good for? In this sense I think its better not to expose the algorithm selection such that it can always be changed internally. As a user of the API I am generally not interested which algorithm is used to solve my problem. What I think is problematic though, that sometimes it can be hard to figure out which algorithm needs to be used, because for certain parameters only certain algorithms work (for example blocked areas etc.).

Reply:
Ok, regarding the parameter setting and I think I better understand this now: we have to do it here to avoid a failure if LM or CH is configured.